### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ruancomelli**
# Description

_Explain briefly about the purpose & context of this PR_

# Screenshots

_If possible add a screenshot and/or gif_

# Example | Usage

_Please share examples and how to reproduce_

---

# Related Issues | Pr

_`close`|`resolve` <ISSUE> (ex. `resolve #20`)_
_link issues to close automatically_

## Summary by Sourcery

Refactor the codebase to improve modularity and introduce new features including password restoration, analytics, and bot management. Enhance validation and error handling, and update the database setup to use MariaDB. Add comprehensive tests for new and existing functionalities.

New Features:
- Introduce a new endpoint `/agent/reset-player-password` allowing agents to reset player passwords.
- Add a new endpoint `/transactions/deposit/pending-coin-transfers` to show total amount of pending coin transfers.
- Implement a new feature for password restoration, including endpoints for password reset and forgot password requests.
- Add support for analytics with endpoints to create, list, and retrieve analytics data.
- Introduce a new bot management feature with endpoints to list and retrieve bot QR codes.

Enhancements:
- Refactor the codebase to separate concerns, splitting the `FinanceServices` into `PaymentServices` and `DepositServices`.
- Improve validation for `movile_number` to ensure it is a numeric string of up to 20 characters.
- Enhance error handling by introducing custom error classes and improving error logging.
- Optimize the deposit verification process by including `clave_rastreo` in query parameters to reduce search complexity.
- Improve the player update process by adding validation and handling for player status changes.

Build:
- Update the Docker setup to use MariaDB instead of MySQL for the database service.

Tests:
- Add comprehensive unit tests for new features such as analytics, bot management, and password restoration.
- Enhance existing tests to cover new validation rules and refactored services.

Chores:
- Update the CHANGELOG to reflect recent changes and new features.
- Remove deprecated migrations and refactor database schema to align with new features.

### CODE REVIEW

**Comment by ruancomelli** [comment]
This is a benchmark review for experiment `mermaid_diagrams`.
Run ID: `mermaid_diagrams/benchmark_2024-09-19T15-23-52_v1-22-0-131-gcca0a8c56`.

This pull request was cloned from `https://github.com/AlphaDev87/timba-api/pull/49`. _(Note: the URL is not a link to avoid triggering a notification on the original pull request.)_
<details>
<summary>Experiment configuration</summary>

```yaml
review_config:
  # User configuration for the review
  # - benchmark - use the user config from the benchmark reviews
  # - <value> - use the value directly
  user_review_config:
    enable_ai_review: true
    enable_rule_comments: false

    enable_complexity_comments: false
    enable_security_comments: false
    enable_tests_comments: false
    enable_comment_suggestions: false
    enable_functionality_review: false

    enable_pull_request_summary: true
    enable_review_guide: true

    enable_approvals: false

  ai_review_config:
    # The model responses to use for the experiment
    # - benchmark - use the model responses from the benchmark reviews
    # - llm - call the language model to generate responses
    model_responses:
      comments_model: benchmark
      comment_area_model: benchmark
      comment_validation_model: benchmark
      comment_suggestion_model: benchmark
      complexity_model: benchmark
      functionality_model: benchmark
      security_model: benchmark
      tests_model: benchmark
      pull_request_summary_model: benchmark
      review_guide_model: llm
      overall_comments_model: benchmark

# The pull request dataset to run the experiment on
pull_request_dataset:
- https://github.com/sourcery-ai/core/pull/4607
- https://github.com/sourcery-ai/core/pull/4631
- https://github.com/sourcery-ai/core/pull/4647
  # CodeRabbit examples:
- https://github.com/2lambda123/-Orange-OpenSource-oorobot/pull/15
- https://github.com/2lambda123/galaxyproject-galaxy/pull/12
- https://github.com/a0v0/avtoolz/pull/79
- https://github.com/adityask98/Hotaru/pull/10
- https://github.com/agdas/vscode/pull/2
- https://github.com/agluszak/hirschgarten/pull/2
- https://github.com/alexsnow348/insightface/pull/46
- https://github.com/alikuxac/utilities/pull/10
- https://github.com/AlphaDev87/timba-api/pull/49
- https://github.com/AngeloTadeucci/Maple2/pull/239
- https://github.com/AngeloTadeucci/Maple2.File/pull/36
- https://github.com/AngeloTadeucci/Maple2/pull/233

# Questions to ask to label the review comments
review_comment_labels: []
# - label: correct
#   question: Is this comment correct?

# Benchmark reviews generated by running
#   python -m scripts.experiment benchmark <experiment_name>
benchmark_reviews: []

```

</details>

**Comment by SourceryAI** [comment]
## Reviewer's Guide by Sourcery

This pull request implements significant changes to the project structure, database schema, and functionality. Key changes include refactoring the transactions system, improving error handling, adding new features like password reset and analytics, and updating various components and services.

### File-Level Changes

| Change | Details | Files |
| ------ | ------- | ----- |
| Refactored transactions system | <ul><li>Split transactions into separate deposits and payments components</li><li>Updated deposit and payment services with new functionality</li><li>Modified database schema for deposits and payments</li><li>Added new endpoints for deposits and payments</li></ul> | `src/components/deposits/services.ts`<br/>`src/components/payments/services.ts`<br/>`src/db/deposits.ts`<br/>`src/db/payments.ts`<br/>`src/routes/transactions/deposits.router.ts`<br/>`src/routes/transactions/payments.router.ts` |
| Improved error handling and logging | <ul><li>Created custom error classes</li><li>Updated error middleware</li><li>Implemented better logging with Logtail</li></ul> | `src/middlewares/errorHandler.ts`<br/>`src/helpers/error/CustomError.ts`<br/>`src/helpers/error/AgentApiError.ts`<br/>`src/helpers/error/AlquimiaApiError.ts`<br/>`src/helpers/loggers.ts` |
| Added password reset functionality | <ul><li>Implemented forgot password and reset password endpoints</li><li>Created password reset token system</li><li>Added email functionality for password reset</li></ul> | `src/components/new-password/controller.ts`<br/>`src/components/new-password/services.ts`<br/>`src/components/new-password/validators.ts`<br/>`src/db/password-reset-token.ts`<br/>`src/helpers/email/email.ts` |
| Implemented analytics system | <ul><li>Created analytics table in database</li><li>Added analytics endpoints</li><li>Implemented analytics controller and services</li></ul> | `prisma/migrations/20240605135628_create_analytics_table/migration.sql`<br/>`src/routes/analytics.router.ts`<br/>`src/components/analytics/controller.ts`<br/>`src/components/analytics/services.ts`<br/>`src/components/analytics/validators.ts` |
| Updated player management | <ul><li>Added functionality to list and update players</li><li>Implemented player password reset by agent</li><li>Updated player validators and services</li></ul> | `src/components/players/controller.ts`<br/>`src/components/players/services.ts`<br/>`src/components/players/validators.ts`<br/>`src/db/players.ts` |
| Improved authentication and authorization | <ul><li>Updated JWT service with configurable token expiration</li><li>Implemented rate limiting for password reset requests</li><li>Added role-based access control to various endpoints</li></ul> | `src/services/jwt.service.ts`<br/>`src/middlewares/rate-limiters/pass-restore-request.ts`<br/>`src/routes/auth.router.ts`<br/>`src/routes/agent.router.ts` |
| Added WhatsApp integration | <ul><li>Implemented WhatsApp notification service</li><li>Added bot QR code management</li></ul> | `src/notification/whatsapp.ts`<br/>`src/components/bot/controller.ts`<br/>`src/components/bot/services.ts`<br/>`src/routes/bot.router.ts` |
| Updated database schema | <ul><li>Added new tables for analytics and chat data</li><li>Modified existing tables structure</li><li>Removed deprecated tables and columns</li></ul> | `prisma/migrations/20240508181836_initial_migration/migration.sql`<br/>`prisma/migrations/20240605135628_create_analytics_table/migration.sql`<br/>`prisma/migrations/20240514142314_create_chat_data_table/migration.sql` |

---

<details>
<summary>Tips</summary>

- Trigger a new Sourcery review by commenting `@sourcery-ai review` on the pull request.
- Continue your discussion with Sourcery by replying directly to review comments.
- You can change your review settings at any time by accessing your [dashboard](https://app.sourcery.ai):
    - Enable or disable the Sourcery-generated pull request summary or reviewer's guide;
    - Change the review language;
- You can always [contact us](mailto:support@sourcery.ai) if you have any questions or feedback.

</details>

File: src/db/players.ts [hunk 1]
```diff
@@ -1,26 +1,73 @@
...
+    try {
+      return prisma.player.count();
+    } catch (e) {
+      throw e;
+    } finally {
+      prisma.$disconnect();
+    }
+  }
+
+  static _getAll = async (
```
Comment (SourceryAI) [inline src/db/players.ts:30]: **suggestion (performance):** Optimize database operations for performance

With the introduction of more granular database operations, ensure that frequently called methods like _getAll are optimized for performance, especially when dealing with large datasets.

File: docker-compose.yml [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
 services:
-  casinomex_db:
-    image: mysql
+  db:
+    image: mariadb
     restart: unless-stopped
     environment:
-      MYSQL_ROOT_PASSWORD: timba2024
-      MYSQL_DATABASE: timba-data-base
+      MARIADB_ROOT_PASSWORD: timba2024root
```
Comment (SourceryAI) [inline docker-compose.yml:7]: **🚨 suggestion (security):** Consider using environment variables for sensitive information

Storing passwords directly in the docker-compose.yml file can be a security risk, especially if the file is committed to version control. Consider using environment variables or a secrets management system for sensitive information.

**Review by SourceryAI — COMMENTED** [review]
Hey @ruancomelli - I've reviewed your changes and they look great!

<details>
<summary>Here's what I looked at during the review</summary>

- 🟡 **General issues**: 1 issue found
- 🟢 **Documentation**: all looks good
</details>

***

[LangSmith trace](https://smith.langchain.com/o/7d8518ad-6dd3-5c7b-96d2-757769296335/projects/p/88560225-5072-48bc-967c-dc54da53f265/r/231f1e4f-98cc-4d46-8338-9e6e0ec72dde?poll=true)

<sub>
Help me be more useful! Please click 👍 or 👎 on each comment to tell me if it was helpful.
</sub>

**Pull request closed without merging**